src/hooks/usePlayground.js

```diff
@@ -0,0 +1,28 @@
+import { useEffect, useState } from 'react';
+import { initialValues } from '../constants';
+import state from '../lib/state';
+import { useAppContext } from '../components/Context';
+import parser from '../parser';
+const savedState = state.load();
+
+function usePlayground() {
+  const [markup, setMarkup] = useState(savedState.markup || initialValues.html);
+  const [query, setQuery] = useState(savedState.query || initialValues.js);
```

Comment:
I think it would be better if the `initialValues` came from the arguments:

```suggestion
function usePlayground({ markup, query }) {
  const [markup, setMarkup] = useState(markup);
  const [query, setQuery] = useState(query);
```

Reply:
I don't think we should do that, since then our context state isn't used anymore and we have to pass `markup`& `query` each time.
So what's the point of getting `markup` & `query` if it's only a local instance then? 🤔

Reply:
That's not how it works though? It's still being provided to the context, it's just the initial value that's passed in?

At least, that's how it now works. So if we create a hook, that's also how the hook should function?

Reply:
`savedState` is the context value, no? 🤔

Reply:
No, saved state comes from the url on page load. After that, it's context what you're mutating all the time, and what all the components read from / write to. One `useEffect` make sure that the state is also pushed back to the URL, but it won't be read from there until you hit <kbd>F5</kbd>

Reply:
Maybe we should be storing the markup and query just in the Context though? The refs are there, so I don't see a good reason to not just add the `markup` and `query` there as well.

Reply:
That's how I thought it was implemented

Reply:
I don't know if it is the case but doing in this way we can have only one context. The idea under the use of the hook was to try to create a playground more than one time. I think that now is not possible to do this, so maybe for now the PR is not really usefull

Reply:
Nothing holds us back from rendering more than one context on a page. So it would still be possible to render multiple playgrounds.

As long as we keep the syncing to URL part out of the hooks and context, this won't be a problem.

Reply:
Syncing the URL with the content is the core of making the playground sharable, like all other REPLs (see Babel, Prettier, ...) do, right? 🤔

Reply:
They're also not having multiple editors on one page, so not sure if we should bother thinking about being able to support that.

Reply:
Also true.

Reply:
Multi editors in one page of course not but maybe in feature the user can save his playgrounds

Reply:
@marcosvega91 If the user could save his playground(s), then we can just update the context value or we could just persist the exact link (like when you would just send it to a friend) and redirect to it